compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -1249,4 +1249,27 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       val CyclicInheritance(symbol, _) :: Nil = messages
       assertEquals("class A", symbol.show)
     }
+
+  @Test def missingCompanionForStatic =
+    checkMessagesAfter("checkStatic") {
+      """
+        |object Foo {
+        |  @annotation.static def bar(): Unit = bar()
+        |}
+      """.stripMargin
+    }.expect { (itcx, messages) =>
+      implicit val ctx: Context = itcx
+      val MissingCompanionForStatic(member) = messages.head
+      assertEquals(member.show, "method bar")
+    }
+
+  @Test def notMissingCompanionForStatic =
```

Comment:
You can remote this test. Here we only test the content of error messages.
We already have tests for correct and incorrect programs: `tests/run/statics.scala`,  `tests/neg/static-no-companion.scala`